public/js/admin.js

```diff
@@ -0,0 +1,46 @@
+;var Admin = (function () {
+    'use strict';
+
+    var $adminDemoContainer = "";
+
+    function setVars() {
+        $adminDemoContainer = $("#adminDemoContainer");
+    }
+
+    function getRandomColor() {
+
+        return '#'+Math.floor(Math.random()*16777215).toString(16);
+    }
+
+    function initAdminPrivateDemo() {
+
+        if ( $adminDemoContainer.length === 0 ) {
+            return;
+        }
+
+        setInterval(function () {
+            $adminDemoContainer.css({
+                "color" : getRandomColor(),
+            });
+        }, 2000);
+    }
+
+    return function () {
+
+        var me = this;
+
+        this.init = function () {
+            setVars();
+            initAdminPrivateDemo();
+        };
+
+        $(document).ready(
+            function () {
+                me.init();
+            }
+        );
+    }
+})();
+
+//init the javascript "class"
+new Admin();
```

Comment:
The goal of using new with a constructor is typically to create an object of a particular type and store that object in a variable. If you don't assign it, or use it, the created object is just thrown away because its reference isn't stored anywhere, and in many cases, this means that the constructor should be replaced with a function that doesn't require new to be used. This rule is aimed at maintaining consistency and convention by disallowing constructor calls using the new keyword that do not assign the resulting object to a variable.

```js
//Bad:
new Thing(); 

//Good:
var thing = new Thing();
Thing();
```